pkg/timeutil/periodic_test.go

```diff
@@ -0,0 +1,189 @@
+			start:    "sat 14:00",
+			duration: "-1h",
+			err:      true,
+		},
+		{ // Too many start fields
+			start:    "sat 14:00 1234",
+			duration: "1h",
+			err:      true,
+		},
+		{ // Specified just a day
+			start:    "sat",
+			duration: "1h",
+			err:      true,
+		},
+	}
+
+	for i, tt := range tests {
+		_, err := ParsePeriodic(tt.start, tt.duration)
+		if err != nil && tt.err == false {
+			t.Errorf("#%d: unexpected error: %q", i, err)
+			continue
+		}
+		if err == nil && tt.err == true {
+			t.Errorf("#%d: expected error and did not get it", i)
+			continue
+		}
+	}
+}
+
+func TestToStart(t *testing.T) {
+	tests := []struct {
+		start    string
+		duration string
+		time     string
+		toStart  time.Duration
+	}{
+		{ // Daily window in 15 minutes.
+			start:    "00:15",
+			duration: "10s",
+			time:     "Thu May 21 00:00:00 PDT 2015",
+			toStart:  15 * time.Minute,
+		},
+		{ // Daily window now.
+			start:    "00:00",
+			duration: "1h",
+			time:     "Thu May 21 00:00:00 PDT 2015",
+			toStart:  0,
+		},
+		{ // Daily window started 1 second ago.
+			start:    "00:00",
+			duration: "1h",
+			time:     "Thu May 21 00:00:01 PDT 2015",
+			toStart:  -1 * time.Second,
+		},
+		{ // Daily window started 10 hours ago but closing edge.
+			start:    "02:33",
+			duration: "10h",
+			time:     "Thu May 21 12:33:00 PDT 2015",
+			toStart:  -10 * time.Hour,
+		},
+		{ // Daily window started 10 hours ago but _just_ closed, expect next.
+			start:    "02:33",
+			duration: "10h",
+			time:     "Thu May 21 12:33:01 PDT 2015",
+			toStart:  13*time.Hour + 59*time.Minute + 59*time.Second,
+		},
+		{ // Daily window started last night but extends into now on the next day.
+			start:    "23:05",
+			duration: "11h",
+			time:     "Thu May 21 09:33:01 PDT 2015",
+			toStart:  -(10*time.Hour + 28*time.Minute + 1*time.Second),
+		},
+		{ // Weekly window started in 15 minutes.
+			start:    "Thu 00:15",
+			duration: "10s",
+			time:     "Thu May 21 00:00:00 PDT 2015",
+			toStart:  15 * time.Minute,
+		},
+		{ // Weekly window now.
+			start:    "Thu 00:00",
+			duration: "1h",
+			time:     "Thu May 21 00:00:00 PDT 2015",
+			toStart:  0,
+		},
+		{ // Weekly window started 1 second ago.
+			start:    "Sun 00:00",
+			duration: "1h",
+			time:     "Sun May 17 00:00:01 PDT 2015",
+			toStart:  -1 * time.Second,
+		},
+		{ // Weekly window started 10 hours ago but closing edge.
+			start:    "Sun 02:33",
+			duration: "10h",
+			time:     "Sun May 17 12:33:00 PDT 2015",
+			toStart:  -10 * time.Hour,
+		},
+		{ // Weekly window started last night in previous week but extends into now on the next day.
+			start:    "Sat 23:05",
+			duration: "11h",
+			time:     "Sun May 17 09:33:01 PDT 2015",
+			toStart:  -(10*time.Hour + 28*time.Minute + 1*time.Second),
+		},
+		{ // Weekly window where next period begins next month
+			start:    "Mon 9:00",
+			duration: "1h",
+			time:     "Sat May 30 23:00:00 PDT 2015",
+			toStart:  34 * time.Hour,
```

Comment:
How about a test where we're in window that started in the last day of the previous month?